internal/bminventory/inventory.go

```diff
@@ -6219,22 +6221,26 @@ func (b *bareMetalInventory) V2DownloadClusterCredentials(ctx context.Context, p
 	fileName := params.FileName
 	respBody, contentLength, err := b.V2DownloadClusterCredentialsInternal(ctx, params)
 
-	// Kubeconfig-noingress has been created during the installation, but it does not have the ingress CA.
-	// At the finalizing phase, we create the kubeconfig file and add the ingress CA.
-	// An ingress CA isn't required for normal login but for oauth login which isn't a common use case.
-	// Here we fallback to the kubeconfig-noingress for the kubeconfig filename.
-	if err != nil && fileName == constants.Kubeconfig {
-		fileName = constants.KubeconfigNoIngress
+	if err != nil {
+		// Kubeconfig-noingress has been created during the installation, but it does not have the ingress CA.
+		// At the finalizing phase, we create the kubeconfig file and add the ingress CA.
+		// An ingress CA isn't required for normal login but for oauth login which isn't a common use case.
+		// Here we fallback to the kubeconfig-noingress for the kubeconfig filename.
+		if fileName == constants.Kubeconfig {
+			fileName = constants.KubeconfigNoIngress
+		}
+		respBody, contentLength, err = b.v2DownloadClusterFilesInternal(ctx, fileName, params.ClusterID.String())
 
-		if b.Config.ClusterTLSCertOverrideDir != "" {
+		if err != nil && (b.installerInvoker == "agent-installer" && b.Config.ClusterTLSCertOverrideDir != "") {
```

Comment:
Could be simplified to

```suggestion
		if err != nil && (b.installerInvoker == "agent-installer") {
```